### PR TITLE
browse: allow filter init via `filter` query param

### LINKED ISSUE #3025: browse: allow filter init via `filter` query param
It would be nice to be able to create links that initialize the file browsing filter so only certain files are displayed by default. For example, if a directory contains many .png and .jpg files, `https://example.com/photos/?filter=.png` would show just the ones with the `.png` substring.

### PR DESCRIPTION
**Pull request by jjlin**
This allows creating links that display only a subset of files in a directory.

Fixes #3025 for Caddy v1.

### CODE REVIEW

**Comment by francislavoie** [comment]
@mholt this was for v1, not the same as the other PR

**Comment by jjlin** [comment]
I thought it would be stating the obvious, so sorry if it wasn't clear. I updated the first comment to clarify the Caddy version being addressed.

File: caddyhttp/browse/setup.go [hunk 1]
```diff
@@ -477,6 +477,16 @@ footer {
 			var filterEl = document.getElementById('filter');
 			filterEl.focus();
 
+			function initFilter() {
+				if (!filterEl.value) {
```
Comment (jjlin) [inline caddyhttp/browse/setup.go]: This check ensures that any user changes to the initial filter value aren't clobbered when navigating forward/backward.

File: caddyhttp/browse/setup.go [hunk 2]
```diff
@@ -477,6 +477,16 @@ footer {
...
+					var filterParam = new URL(window.location.href).searchParams.get('filter');
+					if (filterParam) {
+						filterEl.value = filterParam;
+					}
+				}
+				filter();
```
Comment (jjlin) [inline caddyhttp/browse/setup.go]: `filter()` needs to be executed unconditionally on load to preserve the behavior added in #2368.

**Comment by mholt** [comment]
Oh. Thanks for clarifying.

**Pull request merged**